packages/grid/x-data-grid/src/hooks/features/rows/useGridRowsMeta.ts

```diff
@@ -256,8 +257,8 @@ export const useGridRowsMeta = (
   );
 
   const debouncedHydrateRowsMeta = React.useMemo(
-    () => debounce(hydrateRowsMeta),
-    [hydrateRowsMeta],
+    () => debounce(hydrateRowsMeta, props.rowPositionsDebounceMs),
```

Comment:
I feel like throttle is more suitable instead of debounce here, but changing to throttle didn't cause any visual difference, so I kept it as is.